### PR TITLE
Testing restructure with swr

### PR DESCRIPTION
**Pull request by rclee91**
Related to issue #227 

Added `swr` and `axios` as dependencies
Its either continuing to load or it returns an empty data, as we had discussed.

### CODE REVIEW

File: src/components/pets/PetInfo.js [hunk 1]
```diff
@@ -15,22 +15,40 @@ import { GiAges } from "react-icons/gi";
...
+
+  useEffect(() => {
+    setIsLoading(false);
+    if (!data && !serverError) {
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+    }
+  }, [data]);
+
```
Comment (redxzeta) [inline src/components/pets/PetInfo.js:51]: I think `LoaderComponent` is messing things up so probably be best to remove it. I was trying to be clever haha

```
  const fetcher = (url) =>
    axios.get(url, tokenHeaders).then((res) => res.data.animal);

  const { data, serverError } = useSWR(lookUpPet + id, fetcher);
  const isLoading = !serverError && !data;
  if (serverError) return <h1>Error Loading</h1>;
  if (isLoading)
    return (
      <Spinner animation="grow" variant="primary" role="status">
        <span className="visually-hidden">Loading...</span>
      </Spinner>
    );
    ```

**Review by redxzeta — CHANGES_REQUESTED** [review]
I think it is probably best to not use `LoaderComponent` and `useFetch` since there are issues using with them

File: src/components/pets/PetInfo.js (hunk 1 above)
Comment (rclee91) [reply src/components/pets/PetInfo.js:51]: Alright. I'll change that. Let's see what will happen

**Comment by rclee91** [comment]
Still not giving me data. I'll try it without axios next.

**Comment by rclee91** [comment]
Data is finally loading now for me.
I used it with `fetch()`
I took off `axios` from dependencies
Obviously I'll have to change the tests now cause `useFetch` is no longer used

**Comment by rclee91** [comment]
Changed tests for `PetInfo`
I took out the `fetcher` function to put it as its own component in `utils/petInfoFetcher.js`, so I can create mock implementations for it. 
[This video](https://youtu.be/rvVEWbMkD0A) is what I mostly followed for the tests.

I tried for a long time to use `msw`, for replicating a server similar to what had been done in `testServer.js`, but the I couldn't get it to wait and load data (so basically, no data was returned)

**Review by redxzeta — APPROVED** [review]

**Pull request merged**